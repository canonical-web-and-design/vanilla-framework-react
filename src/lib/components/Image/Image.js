import React from 'react';
import PropTypes from 'prop-types';
import './Image.css';

const Image = (props) => {
  let classArray = [];

  if (props.bordered) {
    classArray = [...classArray, 'p-image--bordered'];
  }
  if (props.shadowed) {
    classArray = [...classArray, 'p-image--shadowed'];
  }

  const classString = classArray.join(' ');

  return <img className={classString} src={props.src} alt={props.alt} />;
};

Image.defaultProps = {
  alt: '',
  bordered: false,
  shadowed: false,
};

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string,
  bordered: PropTypes.bool,
  shadowed: PropTypes.bool,
};

Image.displayName = 'Image';

export default Image;
