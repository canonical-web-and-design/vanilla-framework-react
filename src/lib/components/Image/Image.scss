@import 'vanilla-framework/scss/base';
@import 'vanilla-framework/scss/patterns_image';
@include vf-b-media;
@include vf-p-image;
