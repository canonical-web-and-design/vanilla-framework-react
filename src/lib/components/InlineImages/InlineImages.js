import React from 'react';
import PropTypes from 'prop-types';
import './InlineImages.css';

const InlineImages = (props) => {
  const images = React.Children.map(props.children, child => (
    <li className="p-inline-images__item">
      {child}
    </li>),
  );

  return (
    <ul className="p-inline-images">
      {images}
    </ul>
  );
};

InlineImages.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
  ]).isRequired,
};

InlineImages.displayName = 'InlineImages';

export default InlineImages;
