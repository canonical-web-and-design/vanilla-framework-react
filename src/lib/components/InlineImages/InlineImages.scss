@import 'vanilla-framework/scss/base';
@import 'vanilla-framework/scss/patterns_inline-images';
@import 'vanilla-framework/scss/utilities_clearfix';
@include vf-b-media;
@include vf-p-inline-images;
@include vf-u-clearfix;
