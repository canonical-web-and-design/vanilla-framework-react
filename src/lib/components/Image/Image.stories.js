import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Image from './Image';

storiesOf('Image', module)
  .add('Default',
    withInfo('Default Image component.')(() => (
      <Image src="http://placekitten.com/g/300/300" alt="" />),
    ),
  )
  .add('Bordered',
    withInfo('Bordered Image component.')(() => (
      <Image bordered src="http://placekitten.com/g/300/300" alt="" />),
    ),
  )
  .add('Shadowed',
    withInfo('Image component with drop-shadow.')(() => (
      <Image shadowed src="http://placekitten.com/g/300/300" alt="" />),
    ),
  )
  .add('Bordered and shadowed',
    withInfo('Image component with both border and drop-shadow.')(() => (
      <Image bordered shadowed src="http://placekitten.com/g/300/300" alt="" />),
    ),
  );
