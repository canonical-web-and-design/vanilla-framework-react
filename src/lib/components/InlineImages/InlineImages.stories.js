import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import InlineImages from './InlineImages';
import Image from '../Image/Image';

storiesOf('Inline Images', module)
  .add('Default',
    withInfo("The InlineImages component can be used to showcase a group of related images, such as a group of customer or partner logos. Vanilla's Image components and <img> tags are accepted children.")(() => (
      <InlineImages>
        <Image bordered src="http://placekitten.com/g/200/200" alt="" />
        <Image bordered src="http://placekitten.com/g/400/400" alt="" />
        <Image bordered src="http://placekitten.com/g/300/300" alt="" />
        <Image bordered src="http://placekitten.com/g/500/500" alt="" />
        <Image bordered src="http://placekitten.com/g/800/800" alt="" />
        <img src="http://placekitten.com/g/700/700" alt="" />
      </InlineImages>),
    ),
  );
