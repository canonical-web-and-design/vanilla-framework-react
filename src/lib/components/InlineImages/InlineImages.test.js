import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import InlineImages from './InlineImages';
import Image from '../Image/Image';

describe('InlineImages component', () => {
  it('should render the InlineImages component correctly', () => {
    const inlineImages = ReactTestRenderer.create(
      <InlineImages>
        <Image bordered src="http://placekitten.com/g/200/200" alt="" />
        <Image bordered src="http://placekitten.com/g/400/400" alt="" />
        <Image bordered src="http://placekitten.com/g/300/300" alt="" />
        <Image bordered src="http://placekitten.com/g/500/500" alt="" />
        <Image bordered src="http://placekitten.com/g/800/800" alt="" />
        <Image bordered src="http://placekitten.com/g/700/700" alt="" />
      </InlineImages>);
    const json = inlineImages.toJSON();
    expect(json).toMatchSnapshot();
  });
});
