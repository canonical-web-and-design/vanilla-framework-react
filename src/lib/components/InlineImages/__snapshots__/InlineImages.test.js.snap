// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineImages component should render the InlineImages component correctly 1`] = `
<ul
  className="p-inline-images"
>
  <li
    className="p-inline-images__item"
  >
    <img
      alt=""
      className="p-image--bordered"
      src="http://placekitten.com/g/200/200"
    />
  </li>
  <li
    className="p-inline-images__item"
  >
    <img
      alt=""
      className="p-image--bordered"
      src="http://placekitten.com/g/400/400"
    />
  </li>
  <li
    className="p-inline-images__item"
  >
    <img
      alt=""
      className="p-image--bordered"
      src="http://placekitten.com/g/300/300"
    />
  </li>
  <li
    className="p-inline-images__item"
  >
    <img
      alt=""
      className="p-image--bordered"
      src="http://placekitten.com/g/500/500"
    />
  </li>
  <li
    className="p-inline-images__item"
  >
    <img
      alt=""
      className="p-image--bordered"
      src="http://placekitten.com/g/800/800"
    />
  </li>
  <li
    className="p-inline-images__item"
  >
    <img
      alt=""
      className="p-image--bordered"
      src="http://placekitten.com/g/700/700"
    />
  </li>
</ul>
`;
